lighthouse-core/gather/gatherers/fonts.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+
+/* eslint-disable */
+function getAllLoadedFonts() {
+  const getFont = fontFace => ({
+    display: fontFace.display,
+    family: fontFace.family,
+    stretch: fontFace.stretch,
+    style: fontFace.style,
+    weight: fontFace.weight,
+  });
+
+  return document.fonts.ready.then(() => {
+    return Array.from(document.fonts).filter(fontFace => fontFace.status === 'loaded')
+      .map(getFont);
+  });
+}
+
+function getFontFaceFromStylesheets() {
+  function resolveUrl(url) {
+    const link = document.createElement('a');
```

Comment:
return new URL(url).href
  
this'll canonicalize too

Reply:
I got an error when using return `new URL('./lobster-v20-latin-regular.woff').href` Failed to construct 'URL': Invalid URL

Reply:
then i think you can set the base location and its all good:

```js
return new URL(url, location.href).href
```

Reply:
TIL 👍